spec/calendar.html

```diff
@@ -56,7 +56,10 @@ <h1>CalendarFields ( _calendar_, _fieldNames_ )</h1>
         1. Let _fieldsArray_ be ? CreateArrayFromList(_fieldNames_).
         1. If _fields_ is not *undefined*, then
           1. Set _fieldsArray_ to ? Call(_fields_, _calendar_, « _fieldsArray_ »).
-        1. Return ? CreateListFromArrayLike(_fieldsArray_, « String »).
+        1. Let _result_ be ? IterableToList(_fieldsArray_).
+        1. For each element _name_ of _result_, do
+          1. If Type(_name_) is not String, throw a *TypeError*.
+        1. Return _result_.
```

Comment:
it seems like this should be throwing the error as soon as the item is encountered - in other words, if i pass an infinite iterable where the first item is not a string, i'd expect an exception rather than freezing the thread.

Promise.all (and the other combinators) work this way; i'm not sure if there's a strong reason not to do the validation per-item.

Reply:
Makes sense. I'm not sure how to cleanly do an AO with two independently optional parameters (_elementTypes_ and _method_) so I'm going with a separate operation IterableToListOfType( _items_, _elementTypes_ [ , _method_ ] ), does that sound good?

Reply:
Sure, that works

Reply:
Works for me, though if you aren't going to use the `method` argument you should just leave it out of the AO - no reason to add complexity which isn't used.

Reply:
Updated.